### PR TITLE
Extract fct to eval how many nodes are required

### PR DESCRIPTION
**Pull request by FinchPowers**

### CODE REVIEW

File: starcluster/balancers/sge/__init__.py [hunk 1]
```diff
@@ -749,22 +747,31 @@ def _eval_add_node(self):
...
+        max_add = self.max_nodes - num_nodes
+        return min(self.add_nodes_per_iteration, need_to_add, max_add)
 
+    def _eval_add_node(self):
+        """
+        This function inspects the current state of the SGE queue and decides
+        whether or not to add nodes to the cluster. Returns the number of nodes
+        to add.
+        """
+        need_to_add = self._eval_required_slots_hour()
```
Comment (jraby) [inline starcluster/balancers/sge/__init__.py]: where is this function?
Maybe I'm reading this wrong, but it looks as if `_eval_required_slots_hour`  should be `_eval_required_instances` or the opposite.

**Comment by jraby** [comment]
I like the idea of moving the decision part out of the original function. however I can't really comment on the old or new logic here.

File: starcluster/balancers/sge/__init__.py (hunk 1 above)
Comment (FinchPowers) [inline starcluster/balancers/sge/__init__.py]: _eval_required_instances

**Comment by FinchPowers** [comment]
Merged in datacratic_dev

**Pull request closed without merging**